src/Strategy/Go/GlideLock.hs

```diff
@@ -59,5 +60,9 @@ instance FromJSON GlideLockfile where
 instance FromJSON GlideDep where
   parseJSON = withObject "GlideDep" $ \obj ->
     GlideDep <$> obj .:  "name"
-               <*> obj .: "version"
+               -- version field can be text or an int (hexadecimal hash)
+               <*> (obj .: "version" <|> (intToText <$> obj .: "version"))
```

Comment:
Why are we not just parsing `Text`?  If we're validating input, we can just validate that text is digits or a hex hash, right?

Reply:
This is to handle the overwhelmingly-rare edge case where the hexadecimal number is a valid base-10 int (e.g., `1234` from the test file). In that case, it becomes a yaml `int`, so trying to parse it as `Text` would fail

Reply:
Another damn reason to hate yaml

Reply:
We're just not  p r a g m a t i c  enough. golang's yaml parser will happily coerce numbers to strings! (everything is a string if you squint hard enough)